### ISSUE TITLE
Test failures with bibtexparser 1.1.0

### ISSUE THREAD

**Issue by dotlambda** [pubs/pubs#187]
```
======================================================================
ERROR: test_editor_succeeds_on_second_edit (tests.test_usecase.TestUsecase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/pubs/pubs_cmd.py", line 100, in execute
    args.func(conf, args)
  File "/build/source/pubs/commands/edit_cmd.py", line 61, in command
    metadata=paper.metadata)
  File "/build/source/pubs/paper.py", line 107, in from_bibentry
    bibentry_key, bibdata = bibstruct.get_entry(bibentry)
  File "/build/source/pubs/bibstruct.py", line 39, in get_entry
    verify_bibdata(bibdata)
  File "/build/source/pubs/bibstruct.py", line 33, in verify_bibdata
    raise ValueError("no valid bibdata")
ValueError: no valid bibdata

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/source/tests/test_usecase.py", line 132, in execute_cmds
    _, stdout, stderr = capture_wrap(actual_cmd.split())
  File "/build/source/tests/fake_env.py", line 41, in newf
    return f(*args, **kwargs), _get_fake_stdio_ucontent(sys.stdout), _get_fake_stdio_ucontent(sys.stderr)
  File "/build/source/pubs/pubs_cmd.py", line 103, in execute
    if not uis.get_ui().handle_exception(e):
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  [Previous line repeated 211 more times]
  File "/build/source/pubs/uis.py", line 105, in handle_exception
    self.exit()
  File "/build/source/pubs/uis.py", line 94, in exit
    sys.exit(error_code)
SystemExit: 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/source/tests/test_usecase.py", line 776, in test_editor_succeeds_on_second_edit
    self.execute_cmds(cmds)
  File "/build/source/tests/test_usecase.py", line 152, in execute_cmds
    raise FakeSystemExit(*exc.args).with_traceback(tb)
  File "/build/source/tests/test_usecase.py", line 132, in execute_cmds
    _, stdout, stderr = capture_wrap(actual_cmd.split())
  File "/build/source/tests/fake_env.py", line 41, in newf
    return f(*args, **kwargs), _get_fake_stdio_ucontent(sys.stdout), _get_fake_stdio_ucontent(sys.stderr)
  File "/build/source/pubs/pubs_cmd.py", line 103, in execute
    if not uis.get_ui().handle_exception(e):
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  [Previous line repeated 211 more times]
  File "/build/source/pubs/uis.py", line 105, in handle_exception
    self.exit()
  File "/build/source/pubs/uis.py", line 94, in exit
    sys.exit(error_code)
tests.test_usecase.FakeSystemExit: Exited with code: 1.

======================================================================
ERROR: test_import_ignores (tests.test_usecase.TestUsecase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/pubs/pubs_cmd.py", line 100, in execute
    args.func(conf, args)
  File "/build/source/pubs/commands/import_cmd.py", line 104, in command
    papers = many_from_path(ui, bibpath, ignore=args.ignore_malformed)
  File "/build/source/pubs/commands/import_cmd.py", line 77, in many_from_path
    for k, b in b.items():
AttributeError: 'NoneType' object has no attribute 'items'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/source/tests/test_usecase.py", line 132, in execute_cmds
    _, stdout, stderr = capture_wrap(actual_cmd.split())
  File "/build/source/tests/fake_env.py", line 41, in newf
    return f(*args, **kwargs), _get_fake_stdio_ucontent(sys.stdout), _get_fake_stdio_ucontent(sys.stderr)
  File "/build/source/pubs/pubs_cmd.py", line 103, in execute
    if not uis.get_ui().handle_exception(e):
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  [Previous line repeated 237 more times]
  File "/build/source/pubs/uis.py", line 105, in handle_exception
    self.exit()
  File "/build/source/pubs/uis.py", line 94, in exit
    sys.exit(error_code)
SystemExit: 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/source/tests/test_usecase.py", line 911, in test_import_ignores
    outs = self.execute_cmds(cmds)
  File "/build/source/tests/test_usecase.py", line 152, in execute_cmds
    raise FakeSystemExit(*exc.args).with_traceback(tb)
  File "/build/source/tests/test_usecase.py", line 132, in execute_cmds
    _, stdout, stderr = capture_wrap(actual_cmd.split())
  File "/build/source/tests/fake_env.py", line 41, in newf
    return f(*args, **kwargs), _get_fake_stdio_ucontent(sys.stdout), _get_fake_stdio_ucontent(sys.stderr)
  File "/build/source/pubs/pubs_cmd.py", line 103, in execute
    if not uis.get_ui().handle_exception(e):
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  File "/build/source/tests/fake_env.py", line 87, in handler
    original_handler(ui, exc)
  [Previous line repeated 237 more times]
  File "/build/source/pubs/uis.py", line 105, in handle_exception
    self.exit()
  File "/build/source/pubs/uis.py", line 94, in exit
    sys.exit(error_code)
tests.test_usecase.FakeSystemExit: Exited with code: 1.

======================================================================
FAIL: test_endecodes_raises_exception (tests.test_endecoder.TestEnDecode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/tests/test_endecoder.py", line 170, in test_endecodes_raises_exception
    decoder.decode_bibdata("@misc{I am not a correct bibtex{{}")
AssertionError: BibDecodingError not raised
```

**Comment by dotlambda** [comment]
Btw, I updated bibtexparser to 1.1.0 in the first place because of https://github.com/sciunto-org/python-bibtexparser/issues/225.

**Comment by benureau** [comment]
Indeed, I get the same results with the new bibtexparser. This is probably due to changes on the API of pyparsing.

**Comment by benureau** [comment]
@omangin: bibtexparser returns less exceptions now on malformed data, just returns empty results. Is that expected?

**Comment by benureau** [comment]
@dotlambda: a fix has been pushed to PR #185, and should be merged to master soon.

**Comment by benureau** [comment]
The fix has been merged.
